### ISSUE TITLE
[Ray] Need to chose volume type of head and worker instances

### ISSUE THREAD

**Issue by pomcho555** [ray-project/ray#34570]
### What I did and what I want

I tried to create worker instances as p3.16xlarge on AWS. But I got `InsufficientInstanceCapacity` while Ray head was creating the worker nodes. I tested to spin up a same instance type manually from aws console. And AWS console said "We currently do not have sufficient p3.16xlarge capacity in zones with support for 'gp3' volumes. Our system will be working on provisioning additional capacity." So the root-cause of this error is volume type.

So I suggest a feature to prevent this mess.

### Proposal
Adding a new stanza called `VolumeType`
    This is optional parameter and it would be like this.
    ```yaml
            node_config:
                InstanceType: m5.large
                BlockDeviceMappings:
                    - DeviceName: /dev/sda1
                      Ebs:
                          VolumeSize: 100
                          VolumeType: gp3
    ```

### Misc

Ray `monitor.log`

```
botocore.exceptions.ClientError: An error occurred (InsufficientInstanceCapacity) when calling the RunInstances operation (reached max retries: 0): We currently do not have sufficient p3.16xlarge capacity in the Availability Zone you requested (ap-northeast-1c). Our system will be working on provisioning additional capacity. You can currently get p3.16xlarge capacity by not specifying an Availability Zone in your request or choosing ap-northeast-1a.
```

My configuration file

```yaml
cluster_name: tune-default
provider: { type: aws, region: ap-northeast-1, cache_stopped_nodes: False }
auth: { ssh_user: ec2-user }

min_workers: 2
max_workers: 3
# Customized Deep Learning AMI GPU PyTorch 1.12.1 (Amazon Linux 2) 20230324 with pip and pyenv
available_node_types:
  ray.head.default:
    node_config:
      InstanceType: m5.xlarge
      ImageId: ami-xxxxx
      BlockDeviceMappings:
        - DeviceName: /dev/sda1
          Ebs:
            VolumeSize: 100
  ray.worker.default:
    min_workers: 2
    max_workers: 3
    resources: {}
    node_config:
      InstanceType: p3.16xlarge
      ImageId: ami-001ed5f2a8f0c3e87
      BlockDeviceMappings:
        - DeviceName: /dev/sda1
          Ebs:
            VolumeSize: 100
head_node_type: ray.head.default
setup_commands: # Set up each node.
  - pip install ray torch torchvision tabulate tensorboard pandas wandb
```
  

### Use case

It's gonna be a handy parameter for people like me or who want to choose a specific instance type.

**Comment by zhe-thoughts** [comment]
Assigning to @ gvspraveen to triage. cc @wuisawesome @architkulkarni

**Comment by wuisawesome** [comment]
@pomcho555 You should already be able to use `VolumeType` (though honestly, I'm skeptical that you will actually unlock any additional capacity this way).

```
available_node_types:
  ray.head.default:
    node_config:
      InstanceType: m5.xlarge
      ImageId: ami-xxxxx
      BlockDeviceMappings:
        - DeviceName: /dev/sda1
          Ebs:
            VolumeSize: 100
            VolumeType: io2
```

cc @scv119 @rickyyx

**Comment by pomcho555** [comment]
Thanks, I'll try it out. 
I contacted AWS support about my issue, they also suggested different volume type. I tried gp2, and then it worked for me.

And I also suggest that we document that option in [Cluster YAML Configuration Options
](https://docs.ray.io/en/latest/cluster/vms/references/ray-cluster-configuration.html) I didn't find any description about his. It would be helpful for user like me.
